gnomad_qc/v4/assessment/generate_ld_data.py

```diff
@@ -0,0 +1,566 @@
+from hail.utils import new_temp_file
+
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v3.resources.release import hgdp_tgp_subset
+from gnomad_qc.v4.resources import *
+from gnomad_qc.v4.resources.ld_resources import *
+
+COMMON_FREQ = 0.005
+RARE_FREQ = 0.0005
+
+
+def get_pop_counters(mt, label) -> dict:
+    """
+    Calculate and return genetic ancestry counts given an mt and column label.
+
+    :param mt: Input MT
+    :param label: Col field containing genetic ancestry information
+    :return: Dict of pop counts
+    """
+
+    cut_dict = {
+        f"{label}": hl.agg.filter(
+            hl.is_defined(mt.meta.population_inference.pop)
+            & (mt.meta.population_inference.pop != "oth"),
+            hl.agg.counter(mt.meta.population_inference.pop),
+        ),
+    }
+    cut_data = mt.aggregate_cols(hl.struct(**cut_dict))
+    logger.info(f"Counts: {cut_data}")
+    return cut_data
+
+
+def filter_mt_for_ld(
+    mt, label, pop, common_only: bool = True, re_call_stats: bool = False
+) -> hl.MatrixTable:
+    """
+    Filter MT to only variants and samples appropriate for LD analyses
+
+    :param mt: Input MT to filter
+    :param label: Col field containing genetic ancestry information
+    :param pop: Given genetic ancestry group to filter to
+    :param common_only: Bool of whether to filter to only common variants (>0.5%)
+    :param re_call_stats: Bool to re-calculate callstats. Needed for subsets and when sampling cols.
+    :return: MT filtered to appropriate variants and samples
+    """
+
+    frequency = COMMON_FREQ if common_only else RARE_FREQ
+    # At the moment, ld_matrix OOMs above ~30M variants, so filtering all populations to 0.05% to cut down on variants
+    # All work on standard machines except AFR
+    # Also creating `common_only` ones for most practical uses (above 0.5%)
+    pop_mt = mt.filter_cols(mt.meta.population_inference.pop == pop)
+    meta_index = (
+        hl.enumerate(pop_mt.freq_meta)
+        .find(lambda f: (f[1].get(label) == pop))
+        .collect()[0][0]
+    )
+    # No support for SV datasets in gnomAD v4 Production team code
+    # Have to re-do callstats when noted frequencies don't 100% reflect what you are calculating on
+    # This is for test sets or 1kg_tdpg
+    if re_call_stats:
+        logger.info("Regenerating call_stats...")
+        call_stats = hl.agg.call_stats(pop_mt.GT, pop_mt.alleles)
+        call_stats_bind = hl.bind(
+            lambda cs: cs.annotate(
+                AC=cs.AC[1], AF=cs.AF[1], homozygote_count=cs.homozygote_count[1]
+            ),
+            call_stats,
+        )
+        pop_freq = call_stats_bind
+        pop_mt = pop_mt.annotate_rows(pop_freq=pop_freq)
+
+    else:
+        pop_mt = pop_mt.filter_rows((hl.len(pop_mt.filters) == 0))
+        pop_freq = pop_mt.freq[meta_index]
+        pop_mt = pop_mt.annotate_rows(pop_freq=pop_freq)
+
+    pop_mt = pop_mt.filter_rows((hl.len(pop_mt.filters) == 0))
```

Comment:
More of a note for future reference:
In my case this line of filter is replaced with a set of filters below, 
1. Entries: high quality variants (~VQSR)
2. Entries: adj filters
3. Rows: hl.agg.any(mt.GT.n_alt_alleles() > 0)